test/python/golden/conftest.py

```
@@ -56,14 +57,36 @@ def filter_valid_mesh_shape(system_desc, params, require_exact_mesh=False):
         return num_chips <= num_physical_chips
 
 
+def has_opmodel_enabled():
+    """Check if build was configured with DTTMLIR_ENABLE_OPMODEL=ON"""
+    try:
+        cache_file = os.path.join("build", "CMakeCache.txt")
```

Tests are being run on different machine(s) than build. IMO it is fine that you check whether build has opmodel enabled, however checking build options from cmake cache is not a good idea. It is true that in current CI build dir is preserved as artifact and downloaded before test execution but:
* If someone accidentally change configuration these tests will be just skipped (should fail).
* When we have all releases of wheels (ttrt, alchemist, etc.) we might get rid of upload/download of the whole build directory, and this code will silently skip tests. 
* If a user tries to run tests with release wheel or without having build directory on default location tests won't run.